torch/_subclasses/fake_tensor.py

```diff
@@ -910,10 +915,29 @@ def check_for_subclass(self, flat_arg_tensors):
             for x in flat_arg_tensors
         )
 
-    def check_for_non_fake(self, flat_arg_tensors):
-        return any(
-            isinstance(x, torch.Tensor) and not isinstance(x, FakeTensor)
-            for x in flat_arg_tensors
+    def validate_and_convert_non_fake_tensors(
+        self, flat_arg_tensors, func, converter, args, kwargs
+    ):
+        """
+        Checks if the list of tensors are fake tensors.
+        If not, try to convert them to fake tensors.
+        """
+        for x in flat_arg_tensors:
+            if isinstance(x, torch.Tensor) and not isinstance(x, FakeTensor):
+                if torch.Tag.inplace_view in func.tags:  # type: ignore[attr-defined]
+                    raise Exception(
+                        f"Can't call inplace view ops on global variables or model attributes, Found in {func}(*{args}, **{kwargs})"
```

Comment:
nit: maybe `Can't call metadata mutating ops on non-Fake Tensor Inputs.` ? inplace view is not a user-facing concept, and we might run with non globals/model attirbutes.